tests/parser/features/test_reverting.py

```diff
@@ -1,26 +1,59 @@
+import re
+
 import pytest
+from eth_abi import encode
 from eth_tester.exceptions import TransactionFailed
 
+from vyper.utils import keccak256
+
 pytestmark = pytest.mark.usefixtures("memory_mocker")
 
 
+def build_revert_bytestring(errorDecl, *data):
+    # revert bytes should be 4-byte selector (from keccak of error definition)
+    # followed by abi-encoded data
+    error_selector = keccak256(bytes(errorDecl, "utf-8"))[:4]
+    match = re.search(r"\((.*?)\)", errorDecl)  # .group(1)
+    if match and match.group(1) != "":
+        arg_types = match.group(1)
+        encoded_data = encode(arg_types.split(","), data)
+        return b"".join([error_selector, encoded_data])
+    else:
+        return error_selector
```

Comment:
you probably want abi.encode_single, not abi.encode -- the latter is going to encode them individually, instead of together as a tuple (the encodings are different)

Reply:
That is now deprecated. So use encode wherever possible instead

Reply:
will add a test case encoding some more complex data and see how it handles it

Reply:
honestly, i don't even think it needs to be that complicated, since we don't really need to test arbitrary/generic revert data generation. when i originally said to construct the revert data using utility functions, i was thinking more just like `method_id("NoFives()") + (5).to_bytes(32, "big")`

Reply:
got it, I had to implement my own `method_id` function but otherwise followed what you said and made it simpler